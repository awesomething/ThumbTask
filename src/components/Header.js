//import Home from './Home';
import { Button } from "@material-ui/core";
import React from "react";
//import Home from './Home'
import "./Header.css";
import { Link, useHistory } from "react-router-dom";
//import SearchPage from './SearchPage';
//import FacebookIcon from '@material-ui/icons/Facebook'; <FacebookIcon/>
//import LinkedInIcon from '@material-ui/icons/LinkedIn'; <LinkedInIcon/>
//import React, { useState } from 'react'

function Header() {
  const history = useHistory();
  //const [showSearch, setShowSearch] = useState(false);
  return (
    <div className="flex sm:justify-around justify-between mt-1 mb-2">
      <Link to="/">
        <center>
          <h2 className=" ml-3 text-2xl link">ThumbTask</h2>
        </center>
        <img
          className="header__icon"
          src="https://images.g2crowd.com/uploads/product/image/large_detail/large_detail_bb487be551c0717712203a6d6cd29e01/thumbtack.png"
          alt=""
        />
      </Link>

      <div className="header_info pr-2">
        <h5 className="mb-1">We make writing goals look cool</h5>

        <Button onClick={() => history.push("/searchPage")} variant="outlined">
          Organize Your Tasks
        </Button>
      </div>
    </div>
  );
}

export default Header;
