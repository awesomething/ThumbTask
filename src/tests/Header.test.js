import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Header from '../components/Header'
import {configure} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })


describe(`Header component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
    
  }

  it('renders a Header by default', () => {
    const wrapper = shallow(<Header {...props}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the Header given props', () => {
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
