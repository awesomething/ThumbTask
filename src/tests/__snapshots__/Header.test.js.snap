// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders a Header by default 1`] = `
<div
  className="header"
>
  <Link
    to="/"
  >
    <center>
      <h2>
        ThumbTask
      </h2>
    </center>
    <img
      alt=""
      className="header__icon"
      src="https://images.g2crowd.com/uploads/product/image/large_detail/large_detail_bb487be551c0717712203a6d6cd29e01/thumbtack.png"
    />
  </Link>
  <div
    className="header_info"
  >
    <h5>
      We make writing goals look cool
    </h5>
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
      variant="outlined"
    >
      Organize Your Tasks
    </WithStyles(ForwardRef(Button))>
  </div>
</div>
`;

exports[`Header component renders the Header given props 1`] = `
<div
  className="header"
>
  <Link
    to="/"
  >
    <center>
      <h2>
        ThumbTask
      </h2>
    </center>
    <img
      alt=""
      className="header__icon"
      src="https://images.g2crowd.com/uploads/product/image/large_detail/large_detail_bb487be551c0717712203a6d6cd29e01/thumbtack.png"
    />
  </Link>
  <div
    className="header_info"
  >
    <h5>
      We make writing goals look cool
    </h5>
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
      variant="outlined"
    >
      Organize Your Tasks
    </WithStyles(ForwardRef(Button))>
  </div>
</div>
`;
